app/views/users/show.html.erb

```diff
@@ -1,14 +1,24 @@
 <main class='show'>
+
   <section>
     <h1>Welcome, <%= current_user.username %>!</h1>
     <h3>Your favorite gifs:</h3>
   </section>
+
   <div class='all_gifs'>
-    <% current_user.gifs.each do |gif| %>
-    <div id="img-<%=gif.id%>">
-     <img src=<%= gif.image_path %> class='gif.image_path'>
-     <%= link_to "<i class='far fa-times-circle'></i>".html_safe, gif_path(gif), method: 'put', title: 'Remove from Favorites', class: 'x' %>
-    </div>
+
+    <% current_user.categories.each do |category| %>
+    <h3><%= category.name %></h3>
+    <% category.gifs.each do |gif| %>
+      <% if current_user.gifs.include?(gif) %>
```

Comment:
@alischlereth @iandouglas too much logic in the view? I think I know the answer.... I am having a hard time building out a model method to simplify this. Thoughts?

Reply:
Instead of putting this work on Ruby/Rails, why not do the work in the database by doing a join to only fetch the gifs for that user's categories?